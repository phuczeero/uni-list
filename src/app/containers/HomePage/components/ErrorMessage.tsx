import React, { useEffect, useRef } from 'react';
import styled from 'styled-components/macro';
import { HiExclamation } from 'react-icons/hi';

import { Button } from 'app/components/Button';

interface Props {
  message: string;
  onClose: () => void;
}

export function ErrorMessage(props: Props) {
  const { message, onClose } = props;
  const btnRef = useRef<HTMLButtonElement>(null!);

  useEffect(() => {
    btnRef.current.focus();
  }, []);

  return (
    <Overlay onClick={onClose}>
      <Wrapper>
        <HiExclamation />
        <Message>{message}</Message>
        <Button ref={btnRef} variant="outline" autoFocus>
          Okay
        </Button>
      </Wrapper>
    </Overlay>
  );
}

const Overlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.75);
`;

const Wrapper = styled.div`
  padding: ${p => p.theme.space[3]}px;
  font-size: ${p => p.theme.fontSizes[5]}px;
  text-align: center;
`;

const Message = styled.div`
  color: ${p => p.theme.colors.danger};
`;
