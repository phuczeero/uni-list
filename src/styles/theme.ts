const fonts = {
  body:
    '-apple-system, BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji"',
};
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 72];
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];
const radii = {
  default: 6,
  circle: 99999,
};
const colors = {
  text: '#212529',
  muted: '#dee2e6',
  primary: '#007bff',
  danger: '#dc3545',
  blues: {
    light: '#dcedff',
    dark: '#004896',
  },
};
const sizes = {
  navbar: 64,
  spinner: 32,
};
const shadows = {
  card: '0 0 4px rgba(0, 0, 0, .125)',
};
const breakpoints = ['576px', '768px', '992px', '1200px'];

export const theme = {
  fonts,
  fontSizes,
  space,
  radii,
  colors,
  sizes,
  shadows,
  breakpoints,
};

export type ThemeType = typeof theme;
