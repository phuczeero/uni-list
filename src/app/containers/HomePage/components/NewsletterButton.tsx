import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { homePageActions } from '../slice';
import { selectUniversities } from '../selectors';
import { Button } from 'app/components/Button';

interface Props {}

export function NewsletterButton(props: Props) {
  const universities = useSelector(selectUniversities);
  const dispatch = useDispatch();

  const onClick = () => {
    dispatch(homePageActions.saveToJSON());
  };

  if (universities.length <= 0) {
    return null;
  }

  return <Button onClick={onClick}>Newsletter</Button>;
}
