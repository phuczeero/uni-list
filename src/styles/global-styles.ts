import { createGlobalStyle } from 'styled-components';
import { ThemeType } from './theme';

export const GlobalStyle = createGlobalStyle<{ theme: ThemeType }>`
  * {
    font-family: ${p => p.theme.fonts.body};
    color: ${p => p.theme.colors.text}
  }
`;
