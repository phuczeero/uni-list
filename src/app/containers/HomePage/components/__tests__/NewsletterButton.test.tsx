import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { ErrorMessage } from '../ErrorMessage';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { NewsletterButton } from '../NewsletterButton';
import { Provider } from 'react-redux';
import { Store } from '@reduxjs/toolkit';
import { configureAppStore } from 'store/configureStore';

const renderNewsletterButton = store =>
  render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <NewsletterButton />
      </ThemeProvider>
    </Provider>,
  );

describe('<NewsletterButton />', () => {
  let store: ReturnType<typeof configureAppStore>;
  beforeEach(() => {
    store = configureAppStore();
  });

  it('should not render if universities are empty', () => {
    const mockStore = {
      ...store,
      getState: () => ({
        homePage: {
          universities: [],
        },
      }),
    };
    const component = renderNewsletterButton(mockStore);
    expect(component.container.firstChild).toBe(null);
  });

  it('should render and match the snapshot', () => {
    const mockStore = {
      ...store,
      getState: () => ({
        homePage: {
          universities: [{ name: 'test' }],
        },
      }),
    };
    const component = renderNewsletterButton(mockStore);
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
