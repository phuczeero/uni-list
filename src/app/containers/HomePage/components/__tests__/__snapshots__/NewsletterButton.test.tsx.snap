// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsletterButton /> should render and match the snapshot 1`] = `
.c0 {
  outline: none;
  padding: 8px;
  background-color: #007bff;
  color: white;
  border-radius: 6px;
  border: 2px solid #007bff;
}

.c0:hover,
.c0:focus {
  background-color: #004896;
  border-color: #004896;
}

<button
  class="c0"
>
  Newsletter
</button>
`;
